twisted/internet/interfaces.py

```diff
@@ -2207,6 +2207,28 @@ def clientConnectionForTLS(tlsProtocol):
 
 
 
+class IProtocolNegotiationFactory(Interface):
```

Comment:
Correct. In practice, code that uses this needs to know that you can check `negotiatedProtocol` from your `Protocol` when `dataReceived` is first called.